### PR TITLE
docs: add auth doc

### PR DESCRIPTION
**Pull request by sergebouchut2**
Signed-off-by: Serge Bouchut <serge.bouchut@owkin.com>

## Description

Init documentation on API authentication.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202963105406679

### CODE REVIEW

File: docs/authentication.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    - Payload: contains authentication data (user ID, expiration date) encoded in Base 64.
+    - Signature: the header and the payload encrypted with a secret.
+- The API has a custom layer to split the JWT token in 3 cookies.
+  * A long lifespan cookie which contains the header and the payload.
+  * A short lifespan cookie which contains the signature, using HTTP only (not accessible by JS code to prevent from Cross-Site-Scripting / XSS attacks).
+  * A short lifespan cookie which contains the refreshing expiration date, using HTTP only.
+
+## JWT authentication workflow
+
+1. The client check for existing cookie
```
Comment (jmorel) [inline docs/authentication.md:45]: 1. The client makes a call to `/me/refresh`. 
  * If there was a valid refresh cookie, they get new header+payload and signature cookies.
  * If the call to `/me/refresh` fails (missing or expired refresh cookie), then the client sends login and password to `/me/login` to get all 3 cookies (refresh, header+payload, signature)
2. Client then makes authenticated calls by:
  * sending the signature cookie 
  * and including the contents of the header+payload cookie in a `Authorization: JWT {header+payload}` HTTP header.
4. If this call is rejected for auth reasons, the client tries `/me/refresh` to renew the 2 auth cookies and tries the call again.

**Review by Kelvin-M — COMMENTED** [review]
It can be nice to add `curl`, `python` or `substra` examples for each authentication mode

File: docs/authentication.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  --header 'Content-Type: application/json' \
+  --data '{"username":"org-1","password":"p@sswr0d44"}' \
+  --cookie-jar cookie-jwt.jar \
+  http://substra-backend.org-1.com/me/login/
+{"token_type":"access","exp":1665061851,"iat":1664975451,"jti":"8f6d98b667024db59dcc55f923db8d22","user_id":1}
+
+cat cookie-jwt.jar
+#HttpOnly_.org-1.com    TRUE    /    FALSE    1665580790    refresh           <refresh_value>
+#HttpOnly_.org-1.com    TRUE    /    FALSE    0             signature         <signature_value>
+.org-1.com              TRUE    /    FALSE    1665062390    header.payload    <header_payload_value>
```
Comment (Kelvin-M) [inline docs/authentication.md:119]: A `#` is missing here

**Review by Kelvin-M — APPROVED** [review]
LGTM once comment is resolved

File: docs/authentication.md (hunk 2 above)
Comment (sergebouchut2) [reply docs/authentication.md:119]: No this raw output. Curl add `#` for `HttpOnly`

**Comment by sergebouchut2** [comment]
Force push to trigger PR broken checks

**Pull request merged**